main/webapp/modules/core/scripts/dialogs/management-dialog.js

```diff
@@ -0,0 +1,355 @@
+            });
+            var name = $("<div></div>").addClass("main-text").text(functions[i].name);
+            var expression = $("<div></div>").addClass("sub-text").text(functions[i].expression);
+            td.append(name).append(expression).appendTo(newRow);
+            
+            var actionsCell = $('<td>').css({
+                "border-left" : "none"
+            }).appendTo(newRow);
+    
+            (function (index){
+                $('<button>').text("Edit").addClass("button")
+                .on('click', function (){
+                    self._editFunction(self._column, functionsType, index);
+                }).appendTo(actionsCell);
+    
+                $('<button>').text("Remove").addClass("button")
+                .on('click', function (){
+                    self._deleteFunction(index);
+                })
+                .appendTo(actionsCell);
+            })(i);
+            
+            newRow.appendTo(table);
+        }
+    } else {
+        container.html(
+            '<div style="margin: 1em;"><div style="font-size: 130%; color: #333;">'+"No functions were found currently"+'</div><div style="padding-top: 5px; font-size: 110%; color: #888;">'+"Try adding a new functions below"+'</div></div>'
+        );
+    }
+}
+
+ManagementDialog.prototype._addFunction = function(column) {
+    var self = this;
+    var frame = $(
+        DOM.loadHTML("core", "scripts/dialogs/add-function-dialog.html")
+        .replace("$EXPRESSION_PREVIEW_WIDGET$", ExpressionPreviewDialog.generateWidgetHtml()));
+
+    var elmts = DOM.bind(frame);
+    elmts.dialogHeader.text($.i18n('core-dialogs/add-function'));
+
+    elmts.newFunctionName.text($.i18n('core-dialogs/new-function-name'));
+    elmts.okButton.html($.i18n('core-buttons/ok'));
+    elmts.cancelButton.text($.i18n('core-buttons/cancel'));
+
+    var level = DialogSystem.showDialog(frame);
+    var dismiss = function() { DialogSystem.dismissUntil(level - 1); };
+
+    var o = DataTableView.sampleVisibleRows(column);
+    var previewWidget = new ExpressionPreviewDialog.Widget(
+      elmts, 
+      column.cellIndex,
+      o.rowIndices,
+      o.values,
+      null
+    );
+    
+    elmts.cancelButton.on('click',dismiss);
+    elmts.form.on('submit',function(event) {
+        event.preventDefault();
+        var columnName = jQueryTrim(elmts.functionNameInput[0].value);
+        if (!columnName.length) {
+          alert($.i18n('core-views/warning-function-name'));
+          return;
+        }
+
+        var activeTabName = $("#management-tabs").find(".ui-tabs-active a").text();
+        var add = function() {
+            $.ajax({
+                url: "command/core/get-preference?" + $.param({
+                  name: activeTabName.toLowerCase()
+                }),
+                success: function(data1) {
+                    var _functions = JSON.parse(data1.value);
+                    var langAndExpr = previewWidget.getExpression().split(':');
```

Comment:
Beware that `:` could be contained in the expression itself, so this is not guaranteed to return two elements.